controlplane/eks/api/v1beta1/conversion.go

```diff
@@ -21,25 +21,100 @@ import (
 	infrav1beta1 "sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta1"
 	infrav1beta2 "sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2"
 	ekscontrolplanev1 "sigs.k8s.io/cluster-api-provider-aws/v2/controlplane/eks/api/v1beta2"
-	v1beta2 "sigs.k8s.io/cluster-api-provider-aws/v2/controlplane/eks/api/v1beta2"
 	utilconversion "sigs.k8s.io/cluster-api/util/conversion"
 	"sigs.k8s.io/controller-runtime/pkg/conversion"
 )
 
 // ConvertTo converts the v1beta1 AWSManagedControlPlane receiver to a v1beta2 AWSManagedControlPlane.
 func (r *AWSManagedControlPlane) ConvertTo(dstRaw conversion.Hub) error {
 	dst := dstRaw.(*ekscontrolplanev1.AWSManagedControlPlane)
-
 	if err := Convert_v1beta1_AWSManagedControlPlane_To_v1beta2_AWSManagedControlPlane(r, dst, nil); err != nil {
 		return err
 	}
 
 	// Manually restore data.
 	restored := &ekscontrolplanev1.AWSManagedControlPlane{}
-	if ok, err := utilconversion.UnmarshalData(r, restored); err != nil || !ok {
+	if _, err := utilconversion.UnmarshalData(r, restored); err != nil {
+		return err
+	}
+
+	dst.Spec.IdentityRef = r.Spec.IdentityRef
```

Comment:
Normaly you only need to restore fields here that are new between two APIVersions (i.e. between v1beta1 & v1beta2). For fields that are the same they should be covered by the `autoConvert` and `Convert` functions.

Where you seeing issues?

Reply:
Or was it because of the `AWSManagedControlPlaneClassSpec` in the spec now?

Reply:
Yes, I had to write it manually because of that

Reply:
It seems to be a downside of the code generation, it should support inline nesting replacements. Personally I think the old diff `controlplane/eks/api/v1beta1/zz_generated.conversion.go` can be used as a copy/paste material, but as long as manual diff is the same (it seems to be), it looks ok.